Microsoft.DotNet.Interactive.CSharp/LanguageServiceExtensions.cs

```diff
@@ -23,7 +22,7 @@ public static LinePositionSpan SubtractLineOffset(this LinePositionSpan linePosS
                 linePosSpan.End.SubtractLineOffset(offset));
         }
 
-        public static MarkupContent ToMarkupContent(this QuickInfoItem info)
+        public static string ToMarkdownString(this QuickInfoItem info)
```

Comment:
Should the return type not be `MarkdownContent` instead of `string`?

Reply:
No.  `MarkdownContent` only exists at the LSP serialization layer.  This extension method feeds into `HoverMarkdownProduced` (vs. `HoverPlainTextProduced`) so a regular `string` is appropriate here.  See the modified `HoverTextProduced` type.

Reply:
I think it's useful to have a `MarkdownContent` type in the core domain. The goal of avoiding strings there is so that it's clear what kind of string it is, in any context, not just in the context of an event. 

I wasn't sure if that applies in this part of the code.

Reply:
Having an `abstract ContentBase` caused a bunch of problems with event (de)serialization; there was no way to determine the type.  E.g., consider:

``` C#
abstract class ContentBase
{
    string Value;
}

class PlainTextContent : ContentBase { }

class MarkdownContent : ContentBase { }

class HoverTextProduced
{
    ContentBase Content;
}
```

Then when deserializing the `HoverTextProduced` event from the string `..."content": {"value": "is this plaintext or markdown?"},...`, how is the `content` field to be interpreted?  Without making the serializer much more complicated (1) so it can handle the property that's an abstract class, and (2) so it also carries type information with it (e.g., `"content": {"type":"MarkdownContent", "value": "..."}`).

This felt much cleaner and more in line with the existing (de)serialization work.

Reply:
Yes, a core markdown/plaintext content type would be useful, but is outside the scope of this PR, and given that upcoming work is blocked on this, I thought it best to move forward.  We can revisit/add a backlog item to figure out a markdown editor/builder.